### PR TITLE
Add GritLM

### PR DESCRIPTION
**Pull request by Muennighoff**
https://github.com/embeddings-benchmark/mteb/issues/879

### CODE REVIEW

File: mteb/models/gritlm.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import logging
+from functools import partial
+
+from mteb.model_meta import ModelMeta
+
+logging.basicConfig(level=logging.WARNING)
+logger = logging.getLogger(__name__)
+
+try:
+    from gritlm import GritLM
```
Comment (KennethEnevoldsen) [inline mteb/models/gritlm.py:10]: I would probably pack this in the loader:

```
def gritlm_loader(**kwargs):
    from gritlm import GritLM
    return GritLM(**kwargs)
```

That way you can e.g. check what are all the models trained on.

File: mteb/models/gritlm.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    gritlm7b = ModelMeta(
+        loader=partial(
+            GritLM, "GritLM/GritLM-7B", mode="embedding", torch_dtype="auto"
+        ),
+        name="GritLM/GritLM-7B",
+        languages=[],
```
Comment (KennethEnevoldsen) [inline mteb/models/gritlm.py:17]:
```suggestion
        languages=["eng_Latn", "fra_Latn", "deu_Latn", "ita_Latn", "spa_Latn"]
```

File: mteb/models/gritlm.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        open_source=True,
+        revision="13f00a0e36500c80ce12870ea513846a066004af",
+        release_date="2024-02-15",
+    )
+    gritlm8x7b = ModelMeta(
+        loader=partial(
+            GritLM, "GritLM/GritLM-8x7B", mode="embedding", torch_dtype="auto"
+        ),
+        name="GritLM/GritLM-8x7B",
+        languages=[],
```
Comment (KennethEnevoldsen) [inline mteb/models/gritlm.py:27]:
```suggestion
        languages=["eng_Latn", "fra_Latn", "deu_Latn", "ita_Latn", "spa_Latn"]
```

File: mteb/models/gritlm.py (hunk 1 above)
Comment (Muennighoff) [reply mteb/models/gritlm.py]: Agreed & updated!

**Review by KennethEnevoldsen — APPROVED** [review]
Wonderful!

**Pull request merged**